backend/src/api/game/play.rs

```diff
@@ -1,39 +1,50 @@
-use crate::model::{game_request::GameRequest, game_response::GameResponse};
-use crate::server_error::ServerError;
-use actix_web::web::{self, Json, Path};
+use crate::{
+    api::game::game_state_response::GameStateResponse, db::util::DbPool, model::game::Game,
+    server_error::ServerError,
+};
 use actix_web::post;
-use hive_lib::{game_error::GameError, history::History, position::Position, state::State};
+use actix_web::web::{self, Json, Path};
+use hive_lib::{game_control::GameControl, history::History, position::Position, state::State};
+use serde::Deserialize;
+use serde::Serialize;
+use serde_with::serde_as;
 
-fn get_game_state_from_db(_game_id: u64) -> Result<State, GameError> {
-    let game = "game.txt";
-    let history = History::from_filepath(game)?;
-    State::new_from_history(&history)
+#[serde_as]
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq)]
+pub enum PlayRequest {
+    Turn((String, String)),
+    GameControl(GameControl),
 }
 
-fn play_turn(mut state: State, piece: String, pos: String) -> Result<GameResponse, ServerError> {
-    let board_move = format!("{} {}", piece, pos);
+async fn play_turn(
+    game_id: i32,
+    piece: String,
+    pos: String,
+    pool: &DbPool,
+) -> Result<GameStateResponse, ServerError> {
+    let game = Game::get(game_id, pool).await?;
+    let history = History::new_from_str(game.history.clone())?;
+    let mut state = State::new_from_history(&history)?;
     let piece = piece.parse()?;
     let pos = Position::from_string(&pos, &state.board)?;
     state.play_turn(piece, pos)?;
-    let game = "game.txt";
-    state.history.write_move(game, state.turn, board_move);
-    Ok(GameResponse::new_from_state(&state))
+    GameStateResponse::new_from(&game, &state, pool).await
 }
 
 #[post("/game/{id:\\d+}/play")]
 pub async fn game_play(
-    path: Path<u64>,
-    game_request: Json<GameRequest>,
-) -> Result<Json<GameResponse>, ServerError> {
+    path: Path<i32>,
+    play_request: Json<PlayRequest>,
+    pool: web::Data<DbPool>,
+) -> Result<Json<GameStateResponse>, ServerError> {
     let game_id = path.into_inner();
-    let game_request: GameRequest = game_request.clone(); // This is hacky!
-    let state = get_game_state_from_db(game_id)?;
-    let resp = match game_request {
-        GameRequest::Turn((piece, pos)) => play_turn(state, piece, pos),
-        GameRequest::GameControl(any) => {
+    let resp = match play_request.clone() {
```

Comment:
i don't think `clone()` is needed here, since `play_turn` ends up only needing references to `piece` and `pos` anyway